### PR TITLE
roon-bridge: init at 1.8-795

### PR DESCRIPTION
**Pull request by lovesegfault**
###### Motivation for this change
The implementation was mostly copied from `roon-server` as the packages are very
similar.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 112971` at fba31c7e run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li><a href="https://gist.github.com/8fe2717be71bcfb1a3da23f1467b248f">roon-bridge</a></li></li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 112971` at fba31c7e run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li><a href="https://gist.github.com/173db2394c35b2ee3bf088b40e4d140e">roon-bridge</a></li></li>
  </ul>
</details>

File: pkgs/servers/roon-bridge/default.nix [hunk 1]
```diff
@@ -0,0 +1,64 @@
+{ alsaLib
+, alsaUtils
+, autoPatchelfHook
+, fetchurl
+, lib
+, makeWrapper
+, stdenv
+, zlib
+}: stdenv.mkDerivation rec {
+  name = "roon-bridge";
```
Comment (SuperSandro2000) [inline pkgs/servers/roon-bridge/default.nix:10]:
```suggestion
  pname = "roon-bridge";
```

File: pkgs/servers/roon-bridge/default.nix [hunk 2]
```diff
@@ -0,0 +1,64 @@
+{ alsaLib
+, alsaUtils
+, autoPatchelfHook
+, fetchurl
+, lib
+, makeWrapper
+, stdenv
+, zlib
+}: stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/servers/roon-bridge/default.nix:9]:
```suggestion
}:

stdenv.mkDerivation rec {
```

**Comment by lovesegfault** [comment]
Addressed @SuperSandro2000

**Review by SuperSandro2000 — APPROVED** [review]
package LGTM

File: nixos/modules/services/audio/roon-bridge.nix [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    networking.firewall = mkIf cfg.openFirewall {
+      allowedTCPPortRanges = [
+        { from = 9100; to = 9200; }
+      ];
+      allowedUDPPorts = [ 9003 ];
+    };
+
+
+    users.groups.${cfg.group} = {};
+    users.users.${cfg.user} =
```
Comment (aanderse) [inline nixos/modules/services/audio/roon-bridge.nix:65]: Personally I prefer the pattern where we only provision user and group accounts if `cfg.user` matches the `default` value.

Additionally, if combined with `SupplentaryGroups` would this service work with `DynamicUser`? If the music player intends to manage your music, including the ability to delete, then probably not... but I thought I would ask at least 🤷‍♂️

**Review by aanderse — COMMENTED** [review]
Nice module. Looks good. I left an opinion, but feel free to ignore if you disagree or whatever.

**Comment by aanderse** [comment]
Oh looks like I was late to the party... sorry 😄

**Pull request merged**